dev_scripts_helpers/github/sync_gh_issue_labels.py

```diff
@@ -0,0 +1,326 @@
+#!/usr/bin/env python3
+"""
+Synchronize GitHub issue labels from a label inventory manifest file.
+
+Examples
+
+# Basic usage:
+> sync_gh_issue_labels.py \
+    --yaml ./labels/gh_issues_labels.yml \
+    --owner causify-ai \
+    --repo tutorials \
+    --token *** \
+    --backup
+"""
+
+import argparse
+import logging
+import sys
+from typing import List
+
+import requests
+import yaml
+
+import helpers.hdbg as hdbg
+import helpers.hgit as hgit
+import helpers.hparser as hparser
+import helpers.hsystem as hsystem
+
+_LOG = logging.getLogger(__name__)
+
+
+# #############################################################################
+# Label
+# #############################################################################
+
+
+class Label:
+
+    def __init__(self, name: str, description: str, color: str):
+        """
+        Initialize the label with name, description, and color.
+
+        :param name: label name
+        :param description: label description
+        :param color: label color in hex format
+        """
+        self.name = name
+        self.description = description
+        # Remove # prefix if present.
+        self.color = color.lstrip("#")
+
+    def __repr__(self):
+        return f"Label(name='{self.name}', description='{self.description}', color='{self.color}')"
+
+
+# #############################################################################
+# GitHubClient
+# #############################################################################
+
+
+class GitHubClient:
+
+    def __init__(self, token: str):
+        """
+        Initialize the client with a personal access token.
+
+        :param token: GitHub personal access token
+        """
+        self.token = token
+        self.headers = {
+            "Authorization": f"token {token}",
+            "Accept": "application/vnd.github.v3+json",
+        }
+
+    def get_labels(self, owner: str, repo: str) -> List[Label]:
```

Comment:
We could move owner and repo to the ctor and say "this is a class to operate on a repo", which is the typical use case.

Another approach is to derive a class `LabelProcessor` from `GitHubClient` or better use GitHubClient as a passed element. Composition is always > inheritance